components/inputs/input-radio.js

```diff
@@ -0,0 +1,177 @@
+import { classMap } from 'lit/directives/class-map.js';
+import { FocusMixin } from '../../mixins/focus/focus-mixin.js';
+import { getUniqueId } from '../../helpers/uniqueId.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { InputInlineHelpMixin } from './input-inline-help.js';
+import { PropertyRequiredMixin } from '../../mixins/property-required/property-required-mixin.js';
+import { radioStyles } from './input-radio-styles.js';
+import { SkeletonMixin } from '../skeleton/skeleton-mixin.js';
+
+/**
+ * A radio input within a <d2l-input-radio-group>.
+ * @slot inline-help - Help text that will appear below the input. Use this only when other helpful cues are not sufficient, such as a carefully-worded label.
+ * @slot supporting - Supporting information which will appear below and be aligned with the input.
+ */
+class InputRadio extends InputInlineHelpMixin(SkeletonMixin(FocusMixin(PropertyRequiredMixin(LitElement)))) {
+
+	static get properties() {
+		return {
+			/**
+			 * Checked state
+			 * @type {boolean}
+			 */
+			checked: { type: Boolean, reflect: true },
+			/**
+			 * ACCESSIBILITY: Additional information communicated to screenreader users when focusing on the input
+			 * @type {string}
+			 */
+			description: { type: String },
+			/**
+			 * Disables the input
+			 * @type {boolean}
+			 */
+			disabled: { type: Boolean, reflect: true },
+			/**
+			 * REQUIRED: Label for the input
+			 * @type {string}
+			 */
+			label: { required: true, type: String },
+			/**
+			 * Hides the supporting slot when unchecked
+			 * @type {boolean}
+			 */
+			supportingHiddenWhenUnchecked: { type: Boolean, attribute: 'supporting-hidden-when-unchecked', reflect: true },
+			/**
+			 * Value of the input
+			 * @type {string}
+			 */
+			value: { type: String },
+			_checked: { state: true },
+			_firstFocusable: { state: true },
+			_hasSupporting: { state: true },
+			_isHovered: { state: true },
+			_invalid: { state: true }
+		};
+	}
+
+	static get styles() {
+		return [super.styles, radioStyles, css`
+			:host {
+				display: block;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			.d2l-input-radio-label {
+				cursor: default;
+				margin-block-end: 0;
+			}
+			.d2l-input-inline-help,
+			.d2l-input-radio-supporting {
+				margin-inline-start: 1.7rem;
+			}
+			.d2l-input-radio-supporting {
+				display: none;
+				margin-block-start: 0.6rem;
+			}
+			.d2l-input-radio-supporting-visible {
+				display: block;
+			}
+		`];
+	}
+
+	constructor() {
+		super();
+		this.disabled = false;
+		this.supportingHiddenWhenUnchecked = false;
+		this.value = 'on';
+		this._checked = false;
+		this._firstFocusable = false;
+		this._hasSupporting = false;
+		this._isHovered = false;
+		this._isInitFromGroup = false;
+		this._invalid = false;
+	}
+
+	get checked() { return this._checked; }
+	set checked(value) {
+		if (value === this._checked) return;
+		if (!this._isInitFromGroup) {
```

Comment:
The group manages the value of `checked`, but until the group "sees" this input (via `slotchange`), the input itself needs to look after it.